### PR TITLE
Extend admin workspace search

### PR DESCRIPTION
**Pull request by mads-hartmann**
This extends the admin workspace search so that besides searching for the context url you can for the Workspace ID or Workspace Instance ID.

See issue #3992

### CODE REVIEW

**Comment by mads-hartmann** [comment]
Hey @gtsiolis, can you give me some pointers on how best to fix this UI issue I've introduced? ☺️ Direct link to preview page [here ](https://mads-admin-workspace-search-3992.staging.gitpod-dev.com/admin/workspaces)

Screenshot 2021-04-30 at 12 22 51

**Comment by mads-hartmann** [comment]
Currently blocked by https://github.com/gitpod-io/gitpod/pull/4120

File: components/gitpod-db/src/workspace-db.spec.db.ts [hunk 1]
```diff
@@ -301,5 +301,48 @@ import { DBWorkspaceInstance } from './typeorm/entity/db-workspace-instance';
...
+            expect(dbResult.total).to.eq(1);
+
+            // It should find the workspace with the queried id
+            const workspaceAndInstance = dbResult.rows[0]
+            expect(workspaceAndInstance.workspaceId).to.eq(this.ws.id)
+
+            // It should select the workspace instance that was queried, not the most recent one
+            expect(workspaceAndInstance.instanceId).to.eq(this.wsi1.id)
+        });
+    }
```
Comment (gtsiolis) [inline components/gitpod-db/src/workspace-db.spec.db.ts:346]: **praise**: Haven't tried running these but thanks for adding these tests! 🌟

**Review by gtsiolis — COMMENTED** [review]
> Hey @gtsiolis, can you give me some pointers on how best to fix this UI issue I've introduced?

@mads-hartmann Whoa, somehow I've missed the ping in https://github.com/gitpod-io/gitpod/pull/4111#issuecomment-829999007!🏓

File: components/dashboard/src/admin/WorkspacesSearch.tsx [hunk 2]
```diff
@@ -89,7 +89,7 @@ export function WorkspaceSearch(props: Props) {
                             <path fillRule="evenodd" clipRule="evenodd" d="M6 2a4 4 0 100 8 4 4 0 000-8zM0 6a6 6 0 1110.89 3.477l4.817 4.816a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 010 6z" fill="#A8A29E" />
                         </svg>
                     </div>
-                    <input type="search" placeholder="Search context URLs" onKeyDown={(ke) => ke.key === 'Enter' && search() } onChange={(v) => { setSearchTerm(v.target.value) }} />
+                    <input type="search" placeholder="Search for name, context URL, or instance UUID" onKeyDown={(ke) => ke.key === 'Enter' && search() } onChange={(v) => { setSearchTerm(v.target.value) }} />
```
[the remark this answers is not in the document]
Comment (mads-hartmann) [reply components/dashboard/src/admin/WorkspacesSearch.tsx]: Much better, thanks!

**Comment by mads-hartmann** [comment]
@geropl or @corneliusludmann could one of you have a look please :)

File: components/dashboard/src/admin/WorkspacesSearch.tsx [hunk 3]
```diff
@@ -89,7 +89,7 @@ export function WorkspaceSearch(props: Props) {
                             <path fillRule="evenodd" clipRule="evenodd" d="M6 2a4 4 0 100 8 4 4 0 000-8zM0 6a6 6 0 1110.89 3.477l4.817 4.816a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 010 6z" fill="#A8A29E" />
                         </svg>
                     </div>
-                    <input type="search" placeholder="Search context URLs" onKeyDown={(ke) => ke.key === 'Enter' && search() } onChange={(v) => { setSearchTerm(v.target.value) }} />
+                    <input type="search" placeholder="Search workspaces" onKeyDown={(ke) => ke.key === 'Enter' && search() } onChange={(v) => { setSearchTerm(v.target.value) }} />
```
Comment (corneliusludmann) [inline components/dashboard/src/admin/WorkspacesSearch.tsx:92]: image


NIT: The “Search Users” page has a capitalized user. Would be nice to by in sync here:

```suggestion
                    <input type="search" placeholder="Search Workspaces" onKeyDown={(ke) => ke.key === 'Enter' && search() } onChange={(v) => { setSearchTerm(v.target.value) }} />
```

**Review by corneliusludmann — APPROVED** [review]
Looks good! (one tiny comment inline / not a blocker)

Thanks! :+1:

**Pull request merged**